scalding-core/src/main/scala/com/twitter/scalding/typed/cascading_backend/AsyncFlowDefRunner.scala

```diff
@@ -166,20 +167,23 @@ class AsyncFlowDefRunner(mode: CascadingMode) extends Writer {
       @annotation.tailrec
       def go(id: Long): Unit = messageQueue.take match {
         case Stop => ()
```

Comment:
Can we add the StopFlow stuff here? I think I know what we did wrong: we need to call `go(id)` after we do `flow.stop()` because we need to keep processing messages. We only stop the loop on the Stop message.

Reply:
ahhh yep that fixed it.